GoVet can not jump to err place with go1.13

### What did you do? (required: The issue will be **closed** when not provided)
use  :GoVet
enter on the error line below

I think this error may correspond to the change of the output which `go vet` produce
in before version: go vet will output err message like `./xxx.go:12:2: err msg`,
in 1.13 go the go vet produce `vet: /xxx.go:12:2: err msg`
so when enter on error line it will jump to a "vet: ./xxx.go" place which is invalid

<!--
If possible, please provide clear steps for reproducing the problem.
-->

### What did you expect to happen?
Jump to the go syntax error line
### What happened instead?
showing error message

> Error detected while processing BufEnter Auto commands for "*":
> E344: Can't find directory "vet: ." in cdpath

### Configuration (**MUST** fill this out):
Go 1.13
#### vim-go version:

####  `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):
<details><summary>vimrc</summary><br><pre>

</pre></details>

#### Vim version (first three lines from `:version`):
master
<!-- :version -->

#### Go version (`go version`):
<!-- go version -->
1.13
#### Go environment
<details><summary><code>go env</code> Output:</summary><br><pre>
<!-- go env -->

</pre></details>

yes, that is the point, the problem is jump to the wrong place. I guess the vim-go use `:` and `/` to split the args and use the first item as filepath, when apply to go1.13's `go vet`, the splited first arg become `vet: .` and that is an wrong path
execute `go vet` in the directory which has an error main.go, the outputs are different
go1.10.7's vet
`./main.go:4:2: cannot use _ as value`
go1.13's vet
`vet: ./main.go:4:2: cannot use _ as value or type`

--- a/autoload/go/lint.vim
+++ b/autoload/go/lint.vim
@@ -149,11 +149,11 @@ function! go#lint#Vet(bang, ...) abort
   let l:listtype = go#list#Type("GoVet")
   if l:err != 0
     let l:winid = win_getid(winnr())
-    let errorformat = "%-Gexit status %\\d%\\+," . &errorformat
+    let l:errorformat = "%-Gexit status %\\d%\\+," . &errorformat
     call go#list#ParseFormat(l:listtype, l:errorformat, out, "GoVet")
-    let errors = go#list#Get(l:listtype)
-    call go#list#Window(l:listtype, len(errors))
-    if !empty(errors) && !a:bang
+    let l:errors = go#list#Get(l:listtype)
+    call go#list#Window(l:listtype, len(l:errors))
+    if !empty(l:errors) && !a:bang
       call go#list#JumpToFirst(l:listtype)
     else
       call win_gotoid(l:winid)

--- a/autoload/go/lint_test.vim
+++ b/autoload/go/lint_test.vim
@@ -122,6 +122,35 @@ func! Test_Vet() abort
   endtry
 endfunc
 
+func! Test_Vet_compilererror() abort
+  let l:tmp = gotest#load_fixture('lint/src/vet/compilererror/compilererror.go')
+
+  try
+
+    let expected = [
+          \ {'lnum': 6, 'bufnr': bufnr('%'), 'col': 22, 'valid': 1, 'vcol': 0, 'nr': -1, 'type': '', 'pattern': '', 'text': "missing ',' before newline in argument list (and 1 more errors)"}
+        \ ]
+
+    let winnr = winnr()
+
+    " clear the location lists
+    call setqflist([], 'r')
+
+    call go#lint#Vet(1)
+
+    let actual = getqflist()
+    let start = reltime()
+    while len(actual) == 0 && reltimefloat(reltime(start)) < 10
+      sleep 100m
+      let actual = getqflist()
+    endwhile
+
+    call gotest#assert_quickfix(actual, expected)
+  finally
+    call delete(l:tmp, 'rf')
+  endtry
+endfunc
+
 func! Test_Lint_GOPATH() abort
   let RestoreGOPATH = go#util#SetEnv('GOPATH', fnameescape(fnamemodify(getcwd(), ':p')) . 'test-fixtures/lint')
 

--- a/autoload/go/test-fixtures/lint/src/vet/compilererror/compilererror.go
+++ b/autoload/go/test-fixtures/lint/src/vet/compilererror/compilererror.go
@@ -1,0 +1,7 @@
+package main
+
+import "fmt"
+
+func main() {
+	fmt.Println("vim-go"
+}

--- a/compiler/go.vim
+++ b/compiler/go.vim
@@ -31,14 +31,13 @@ endif
 " use a different output, for those we define them directly and modify the
 " errorformat ourselves. More information at:
 " http://vimdoc.sourceforge.net/htmldoc/quickfix.html#errorformat
-CompilerSet errorformat =%-G#\ %.%#                   " Ignore lines beginning with '#' ('# command-line-arguments' line sometimes appears?)
-CompilerSet errorformat+=%-G%.%#panic:\ %m            " Ignore lines containing 'panic: message'
-CompilerSet errorformat+=%Ecan\'t\ load\ package:\ %m " Start of multiline error string is 'can\'t load package'
-CompilerSet errorformat+=%A%f:%l:%c:\ %m              " Start of multiline unspecified string is 'filename:linenumber:columnnumber:'
-CompilerSet errorformat+=%A%f:%l:\ %m                 " Start of multiline unspecified string is 'filename:linenumber:'
-CompilerSet errorformat+=%C%*\\s%m                    " Continuation of multiline error message is indented
-CompilerSet errorformat+=%-G%.%#                      " All lines not matching any of the above patterns are ignored
-
+CompilerSet errorformat =%-G#\ %.%#                                 " Ignore lines beginning with '#' ('# command-line-arguments' line sometimes appears?)
+CompilerSet errorformat+=%-G%.%#panic:\ %m                          " Ignore lines containing 'panic: message'
+CompilerSet errorformat+=%Ecan\'t\ load\ package:\ %m               " Start of multiline error string is 'can\'t load package'
+CompilerSet errorformat+=%A%\\%%(%[%^:]%\\+:\ %\\)%\\?%f:%l:%c:\ %m " Start of multiline unspecified string is 'filename:linenumber:columnnumber:'
+CompilerSet errorformat+=%A%\\%%(%[%^:]%\\+:\ %\\)%\\?%f:%l:\ %m    " Start of multiline unspecified string is 'filename:linenumber:'
+CompilerSet errorformat+=%C%*\\s%m                                  " Continuation of multiline error message is indented
+CompilerSet errorformat+=%-G%.%#                                    " All lines not matching any of the above patterns are ignored
 let &cpo = s:save_cpo
 unlet s:save_cpo
 

